SourceCode/Transport/ftt_tr_main.py

```diff
@@ -218,21 +208,25 @@ def solve(data, time_lag, iter_lag, titles, histend, year, specs):
             # Energy demand (BTTC)           MJ/km
 
             data['TJEF'][r, :, 0] = (np.matmul(np.transpose(fuel_converter), data['TEWG'][r, :, 0]*\
-                                    data['BTTC'][r, :, c3ti['9 energy use (MJ/km)']]*CO2Corr[r]/41.868))
+                                    data['BTTC'][r, :, c3ti['9 energy use (MJ/km)']]*CO2_corr[r]/41.868))
 
             # "Emissions"
             data['TEWE'][r, :, 0] = (data['TEWG'][r, :, 0] * data['BTTC'][r, :, c3ti['14 CO2Emissions']]
-                                     * CO2Corr[r] * emis_corr[r,:]/1e6)
-
+                                     * CO2_corr[r] * emis_corr[r,:]/1e6)
+    
+    # Call the survival function routine, updating scrappage and age matrix:
+    if year <= np.max(data["TDA1"][:, 0, 0]):
+        data = survival_function(data, time_lag, histend, year, titles)
+        
 
     # Calculate the LCOT for each vehicle type.
     # Call the function
     data = get_lcot(data, titles, year)
 
     # %% Simulation of stock and energy specs
 
-    if year > data["TDA1"][r, 0, 0]:
-        # Note, regions have different start dates (TDA1) 
+    if year > np.min(data["TDA1"][:, 0, 0]): 
```

Comment:
I believe that code was wrong before. At this point in the code, we're just past an r loop. So this would always use r=71. The code skips over some regions in the time step loop later. I've added a note to the comments to explain.